packages/custom_lint/lib/src/v2/server_to_client_channel.dart

```diff
@@ -503,7 +488,7 @@ $dependencies
     await Future.wait([
       _tempDirectory.delete(recursive: true),
       _serverSocket.then((value) => value.close()),
-      _process.future.then((value) => value.kill()),
+      if (_process.isCompleted) _process.future.then((value) => value.kill()),
```

Comment:
Why this?

Reply:
Well the `Completer` should have been completed in `init`. If it wasn't because of errors thrown before it was completed, then this will wait forever (the hanging).

> There is no need to complete the Completer with the errors from the Process, since it is awaited they are also thrown to the init() caller, completing the Completer<Process> with the error would cause close() to rethrow the error again, this would cause further problems in the plugin that calls close.

And further because of this, we don't need the errors in the `Completer`, this is what the funky future did before.

Reply:
I've added an onError instead. It probably makes more sense.  
And removed the complete for a `late final Future<T>`. So if somehow we forget to assign the future, `close` would throw instead of hang.

Reply:
Yea I guess, was thinking the same but thought you must have had your reasons for the completer.